octoprint_mrbeam/static/js/app/view-models/material-store.js

```diff
@@ -36,12 +41,25 @@ $(function () {
             } else {
                 self.showConnectionError();
             }
+            self.mrBeamVM.removeNotifyIcon($("#materialstore_tab_btn"));
         };
 
         self.sendInitDetailsToMaterialStoreIframe = function () {
             self.sendMessageToMaterialStoreIframe(
                 MATERIAL_STORE_EVENT_TYPE.MR_BEAM_LOAD
             );
+
+            // send new laserhead model ID if changed
+            if (self.laserheadChangedVM.laserheadXDetected()) {
+                self.sendMessageToMaterialStoreIframe(
+                    MATERIAL_STORE_EVENT_TYPE.SELECT_FILTER,
+                    {
+                        type: "recommendedBy",
+                        subsection: "laserheadModel",
+                        value: "x",
```

Comment:
On the material store it will be an actual filter, not recommendation and as far as I know we can only filter by "tags", "categories" or "metadata". I think on the mrbeam plugin we don't need to know that so we could just send the name of the filter like: value: 'laserheadx'. We could keep an enum with the expected available filters for the material store on the Mr Beam plugin, sort of what we have with MATERIAL_STORE_EVENT_TYPE. 
Shall we talk about it?

Reply:
done